src/tests/functional/shared_test_classes/src/base/low_precision_transformations/layer_transformation.cpp

```diff
@@ -127,6 +127,32 @@ std::string LayerTransformation::get_runtime_precision_by_fused_name(const std::
     return find_node_by_runtime_precision(compiledModel, is_node_f);
 }
 
+bool LayerTransformation::check_execution_order(const std::vector<std::string>& layerTypes) {
+    size_t i = 0;
+    const std::shared_ptr<const ov::Model>& execFunction = compiledModel.get_runtime_model();
+    for (const auto& op : execFunction->get_ordered_ops()) {
+        const auto& rtInfo = op->get_rt_info();
+        const auto& typeIt = rtInfo.find("layerType");
+        OPENVINO_ASSERT(typeIt != rtInfo.end(), "Layer is not found");
+
+        const auto layerType = typeIt->second.as<std::string>();
+        if (i == 0) {
+            if (layerTypes[0] == layerType) {
+                // start comparing
+                i++;
+            }
+        } else {
+            if (layerTypes[i] != layerType) {
+                return false;
+            }
+            i++;
+        }
+    }
+
+    // all types are compared
+    return i == layerTypes.size();
+}
```

Comment:
It seems like the current realization fails in case when `layerTypes` is empty. Also, I think we could refactor it using `std::find_if`. Smth like that:
```
bool LayerTransformation::check_execution_order(const std::vector<std::string>& layerTypes) {
    const auto ops = compiledModel.get_runtime_model()->get_ordered_ops();
    auto found_it = ops.begin();
    for (const auto& refLayerType : layerTypes) {
        auto layer_matches = [&refLayerType](const std::shared_ptr<ov::Node>& op) {
            const auto& rtInfo = op->get_rt_info();
            const auto& typeIt = rtInfo.find("layerType");
            OPENVINO_ASSERT(typeIt != rtInfo.end(), "Layer is not found");

            const auto layerType = typeIt->second.as<std::string>();
            return layerType == refLayerType;
        };
        found_it = std::find_if(found_it, ops.end(), layer_matches);
        if (found_it == ops.end())
            return false;
    }
    return true;
}
```

Reply:
Thanks for comment. Operation types sequence is checked here. If first operation is used, then all operations have to be from the list. It's a reason why operation list is ordered. But operation types collection dosn't consist all operations from an execution graph to avoid influence on other functionality. It's a reason why "start comparing" is neccessary.

Reply:
additional comment was added, thanks